components/Front/Link.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { Link } from '../../lib/routes'
+import { parse } from 'url'
+
+import { PUBLIC_BASE_URL } from '../../lib/constants'
+import ArticleLink from '../Feed/ArticleLink'
+
+const PUBLIC_HOSTNAME = parse(PUBLIC_BASE_URL).hostname
+
+// ToDo:
+// consider setting window.location for
+// passHref false and external / unrecognized links
+// for those links currently only the headline is clickable
+
+export default ({ href, passHref, children }) => {
+  if (!href) {
+    return children
+  }
+  const urlObject = parse(href)
+  if (urlObject.hostname && urlObject.hostname !== PUBLIC_HOSTNAME) {
+    // do nothing if url has a hostname and it's not ours
+    return children
+  }
+  // path or url is ours
```

Comment:
I'm generally OK with pattern matching external vs. internal links and logic based on that. Some of this might have to move into util methods at some point, but let's wait for the first 2nd use case to DRY things up.